JEasyCrypto/src/easycrypto/ROT13Method.java

```diff
@@ -0,0 +1,39 @@
+package easycrypto;
+
+import easycrypto.EasyCryptoAPI.Result;
+import easycrypto.EasyCryptoAPI.ResultCode;
+
+class ROT13Method implements CryptoMethod {
+
+    @Override
+    public Result encrypt(final String toEncrypt) {
+        String toStoreTo = toRot13(toEncrypt);
+        return new Result(ResultCode.ESuccess, toStoreTo);
+    }
+
+    @Override
+    public Result decrypt(final String toDecrypt) {
+        String toStoreTo = toRot13(toDecrypt);
+        return new Result(ResultCode.ESuccess, toStoreTo);
+    }
+
+    @Override
+    public String method() {
+        return "rot13";
+    }
+
+    private String toRot13(final String input) {
```

Comment:
We should also handle other inputs (numbers, Unicode etc.).

Reply:
E.g. printing error message to the user!

Reply:
@mbuet2ner ROT13 doesn't actually take numbers into account - it leaves them as they are (which is actually correct behavior). For example ROT18 would handle numbers, but ROT13 doesn't because it only rotates alphabets from a to z. Same for unicode characters (they should just pass through as they are, it seems like we need to fix that one).

About the error handling, do you want the error message to be shown whenever the algorithm didn't rotate a specific character? We feel that it shouldn't as not rotating a character other than a-z/A-Z is not an error in ROT13. We tried other ROT13 converters and they all passed the other characters through without errors.

Reply:
Yes and precisely that is why we maybe should validate that only alpha characters were entered (e.g. using regex) or at least tell the user beforehand, that numbers will *not* be encrypted.

It is certainly not an error in ROT13. But I think we should anticipate "wrong" user input and handle that accordingly, since it is a best practice in software engineering.

Reply:
Ok, thanks for the comments. A few further questions for clarification before we implement the fixes:

> we maybe should validate that only alpha characters were entered

Does this mean that input with any non-alphabetic characters would not be allowed at all? That sounds quite restrictive (at least in our opinion). I would personally prefer showing just the warning and passing any non-alphabetic characters as they are. Which one would you prefer?

> I think we should anticipate "wrong" user input and handle that accordingly

What would "wrong" input be in this case, non-alphabetic characters? Would this need other handling than displaying the warning message?

> E.g. printing error message to the user!

Would it be an error message or a warning message? An error message would indicate that the "encryption" will not be done at all

Reply:
Warning message sounds good to me. 

> I would personally prefer showing just the warning and passing any non-alphabetic characters as they are.

Sounds also reasonable. As long as the user knows about it, I think we are covered.

Reply:
Now a warning message is shown when the user selects the rot13 method.

Reply:
Thanks!